public/app/plugins/datasource/prometheus/components/PromExploreExtraField.tsx

```diff
@@ -4,17 +4,21 @@ import { css, cx } from 'emotion';
 
 // Types
 import { InlineFormLabel, RadioButtonGroup } from '@grafana/ui';
+import { PromQuery } from '../types';
+import { PromExemplarField } from './PromExemplarField';
 
 export interface PromExploreExtraFieldProps {
   queryType: string;
   stepValue: string;
+  query: PromQuery;
   onStepChange: (e: React.SyntheticEvent<HTMLInputElement>) => void;
   onKeyDownFunc: (e: React.KeyboardEvent<HTMLInputElement>) => void;
   onQueryTypeChange: (value: string) => void;
+  onChange: (value: PromQuery) => void;
```

Comment:
I would name this `onExemplarsChange` as we have also other onChange props (`onQueryTypeChange, onStepChange`). So it is more clear.

Reply:
I'm not sure, this is the onChange handler for the query editor. Maybe it would make sense to move the on change handlers to this component and only pass the onChange and query props down? What do you think?

Reply:
>Maybe it would make sense to move the on change handlers to this component and only pass the onChange and query props down? What do you think?

Yes, I 🧡 the idea When I was refactoring & adding the query type picker to Loki, I did the same thing for [LokiOptionFields](https://github.com/grafana/grafana/blob/master/public/app/plugins/datasource/loki/components/LokiOptionFields.tsx), which is basically same as PrometheusExploreExtraField. But it defnitely doesn't have to be part of this PR, we can refactor it after this is merged, separately.